core/src/main/java/org/mule/runtime/core/internal/exception/ErrorHandler.java

```diff
@@ -76,6 +81,26 @@ public CoreEvent handleException(Exception exception, CoreEvent event) {
     throw new MuleRuntimeException(createStaticMessage(MUST_ACCEPT_ANY_EVENT_MESSAGE));
   }
 
+  @Override
+  public void routeMessagingError(Exception error, Consumer<CoreEvent> handledConsumer,
+                                  Consumer<Throwable> failureConsumer) {
+    MessagingException messagingError = (MessagingException) error;
+    CoreEvent event = messagingError.getEvent();
+    messagingError.setProcessedEvent(event);
+    try {
+      for (MessagingExceptionHandlerAcceptor errorListener : exceptionListeners) {
+        if (errorListener.accept(event)) {
+          errorListener.routeMessagingError(error, handledConsumer, failureConsumer);
+          return;
+        }
+      }
+      throw new MuleRuntimeException(createStaticMessage(MUST_ACCEPT_ANY_EVENT_MESSAGE));
+    } catch (Exception e) {
+      failureConsumer.accept(new MessagingExceptionResolver(this).resolve(new MessagingException(event, e, this),
```

Comment:
have the `new MessagingExceptionResolver(this)` at the instance, instead of building it for every error handled